x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinatorTests.java

```diff
@@ -2350,6 +2353,94 @@ public void testFollowerIndexIsCreatedInExecuteAutoFollow() {
         assertThat(autoFollowResults.v2().contains(indexName), equalTo(true));
     }
 
+    public void testRemovesStatsOnRemovingAutoFollowPattern() {
+        // given auto-follow pattern added
+        var pattern1 = createAutoFollowPattern("remote1", "logs-*");
+        var pattern2 = createAutoFollowPattern("remote2", "logs-*");
+        var pattern3 = createAutoFollowPattern("remote2", "metrics-*");// same remote
+
+        var autoFollowCoordinator = createAutoFollowCoordinator();
+        autoFollowCoordinator.updateAutoFollowers(
+            createClusterStateWith(Map.of("pattern1", pattern1, "pattern2", pattern2, "pattern3", pattern3))
+        );
+
+        // and stats are published
+        autoFollowCoordinator.updateStats(
+            List.of(
+                new AutoFollowCoordinator.AutoFollowResult("pattern1", new RuntimeException("ClusterStateFetchException")),
+                new AutoFollowCoordinator.AutoFollowResult(
+                    "pattern1",
+                    List.of(Tuple.tuple(new Index("logs-1", UUIDs.base64UUID()), new RuntimeException("AutoFollowExecutionException")))
+                ),
+                new AutoFollowCoordinator.AutoFollowResult("pattern2", new RuntimeException("ClusterStateFetchException")),
```

Comment:
I think we should also test the case where there is no outer level exception, only an exception for a specific index (since that is the "normal" case).